app/src/main/java/com/iitp/njack/iitp_connect/core/youtube/PlaylistRepository.java

```diff
@@ -0,0 +1,139 @@
+package com.iitp.njack.iitp_connect.core.youtube;
+
+import android.app.Activity;
+import android.arch.lifecycle.MutableLiveData;
+import android.content.Context;
+import android.os.AsyncTask;
+import android.util.Log;
+import com.google.api.client.extensions.android.http.AndroidHttp;
+import com.google.api.client.googleapis.extensions.android.gms.auth.GoogleAccountCredential;
+import com.google.api.client.googleapis.extensions.android.gms.auth.UserRecoverableAuthIOException;
+import com.google.api.client.http.HttpTransport;
+import com.google.api.client.json.JsonFactory;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import com.google.api.services.youtube.YouTube;
+import com.google.api.services.youtube.model.Channel;
+import com.google.api.services.youtube.model.ChannelListResponse;
+import com.google.api.services.youtube.model.Playlist;
+import com.google.api.services.youtube.model.PlaylistListResponse;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import javax.inject.Singleton;
+
+public class PlaylistRepository {
+    Context context;
+    private List<YoutubePlaylist> youtubePlaylists = new ArrayList<>();
+    private GoogleAccountCredential googleAccountCredential;
+    private MutableLiveData<List<YoutubePlaylist>> playlists = new MutableLiveData<>();
+
+
+    PlaylistRepository(GoogleAccountCredential googleAccountCredential) {
+        this.googleAccountCredential = googleAccountCredential;
+        getDataFromAPI(context);
+    }
+
+    MutableLiveData<List<YoutubePlaylist>> getPlaylists() {
+        return this.playlists;
+    }
+
+    void getDataFromAPI(Context context) {
+        this.context = context;
+        new MakeRequestTask(googleAccountCredential).execute();
+        this.youtubePlaylists.add(new YoutubePlaylist("abcd"));
+
+        playlists.setValue(youtubePlaylists);
+    }
+
+    @Singleton
+    private class MakeRequestTask extends AsyncTask<Void, Void, List<YoutubePlaylist>> {
```

Comment:
Don't use AsyncTasks. Use RxJava. Async Tasks are not used in the modern android applications.